libs/stanag4609/src/main/java/org/codice/alliance/libs/stanag4609/AbstractMetadataPacket.java

```diff
@@ -63,12 +72,25 @@ private boolean validateChecksum(final KlvContext klvContext, final byte[] klvBy
                 "Decoded KLV packet didn't contain checksum (which is required).");
     }
 
-    protected final byte[] getPESPacketPayload(final int packetLength,
-            final int expectedHeaderLength) {
+    protected final byte[] getPESPacketPayload() {
+
+        if (this.pesPacketBytes.length < BASE_PES_PACKET_HEADER_LENGTH) {
+            return null;
+        }
+
+        int additionalHeaderBytes = this.pesPacketBytes[PES_HEADER_LENGTH_INDEX];
```

Comment:
I'm not 100% sure but I think the PES header length byte is an unsigned value. We could do `Byte.toUnsignedInt(this.pesPacketBytes[PES_HEADER_LENGTH_INDEX])` to handle that.